src/main/scala/org/constellation/p2p/Cluster.scala

```diff
@@ -463,18 +461,35 @@ class Cluster[F[_]: Concurrent: Logger: Timer: ContextShift](ipManager: IPManage
       "cluster_leave"
     )
 
-  def broadcastNodeState(): F[Unit] =
+  def broadcastNodeState(nodeState: NodeState): F[Unit] =
```

Comment:
can we use current `nodeState` instead of provided one by argument? it can lead to misalignment (different current state than broadcasted)

Reply:
since nodesState can alter can also happen in the middle of 2 functions it's also not a solution,I will make it private so it is only used in state transition function

Reply:
but then you'll have different current state than broadcasted one, won't you?

Reply:
state can change over time, we always broadcast all state changes (except from one that it's not necessary) previously losing state transition was possible now it's not